src/TLuaInterpreter.cpp

```diff
@@ -15449,6 +15450,69 @@ int TLuaInterpreter::getConnectionInfo(lua_State *L)
     return 2;
 }
 
+int TLuaInterpreter::unzipAsync(lua_State *L)
+{
+    QString zipLocation;
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "unzipAsync: bad argument #1 type (zip location as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+    zipLocation = QString::fromUtf8(lua_tostring(L, 1));
+
+    QString extractLocation;
+    if (!lua_isstring(L, 2)) {
+        lua_pushfstring(L, "unzipAsync: bad argument #2 type (extract location as string expected, got %s!)", luaL_typename(L, 2));
+        return lua_error(L);
+    }
+    extractLocation = QString::fromUtf8(lua_tostring(L, 2));
+
+    QTemporaryDir temporaryDir;
+    if (!temporaryDir.isValid()) {
+        lua_pushnil(L);
+        lua_pushstring(L,
+                       "couldn't create temporary directory to extract the zip into");
+        return 2;
+    }
+
+    extractLocation = QDir::fromNativeSeparators(extractLocation);
+    if (!extractLocation.endsWith(QLatin1String("/"))) {
+        extractLocation.append(QLatin1String("/"));
+    }
+
+    QDir dir;
+    if (!dir.mkpath(extractLocation)) {
+        lua_pushnil(L);
+        lua_pushstring(L,
+                       "couldn't create output directory to extract the zip into");
```

Comment:
:thought_balloon: This message is very similar to the previous one which sort of sounds like the extraction is taking place twice - which isn't the case - so would it be more reasonable to use:
```CPP
    "couldn't create output directory to put the extracted files into"
```
here ?

Reply:
Sure